packages/neon/neon_files/lib/models/file_details.dart

```diff
@@ -30,4 +76,10 @@ class FileDetails {
   final bool? hasPreview;
 
   final bool? isFavorite;
+
+  final Stream<double>? progress;
+  final bool isUploading;
+  final bool isDownloading;
+
+  bool get isLoading => isUploading || isDownloading;
```

Comment:
I've read the code further down and I think it would make sense to use a nullable enum for this. That's way easier to understand and using 2 booleans is a bit ugly especially since they are exclusive of each other.